### PR TITLE
[DDL] Minor refactor of the ddl pkg

### PR DESCRIPTION
**Pull request by Tang8330**
## Changes

1. Adding a debug log to dropping temporary tables
2. Changing `AlterTable` to a receiver method

### CODE REVIEW

File: clients/bigquery/bigquery.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func (s *Store) PrepareTemporaryTable(tableData *optimization.TableData, tableCo
 			Mode:              tableData.Mode(),
 		}
 
-		if err := ddl.AlterTable(tempAlterTableArgs, tableData.ReadOnlyInMemoryCols().GetColumns()...); err != nil {
+		if err := tempAlterTableArgs.Alter(tableData.ReadOnlyInMemoryCols().GetColumns()...); err != nil {
```
Comment (nathan-artie) [inline clients/bigquery/bigquery.go:52]: I feel like dropping `Table` calling the function just `Alter` makes it hard to tell what is being altered, and now that it's a receiver function on an argument struct if I were seeing it for the first time I would assume that this code was just altering the `tempAlterTableArgs` struct, not creating a temporary table.

Comment (Tang8330) [reply clients/bigquery/bigquery.go:52]: Oh interesting. For me, I look at `ALTER` and automatically assume it's the SQL keyword for DDL. I can add `AlterTable` back to be more clear though

File: lib/destination/ddl/ddl.go [hunk 2]
```diff
@@ -23,7 +23,9 @@ import (
 // Temporary tables look like this: database.schema.tableName__artie__RANDOM_STRING(5)_expiryUnixTs
 func DropTemporaryTable(dwh destination.DataWarehouse, fqTableName string, shouldReturnError bool) error {
 	if strings.Contains(strings.ToLower(fqTableName), constants.ArtiePrefix) {
-		_, err := dwh.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", fqTableName))
+		sqlCommand := fmt.Sprintf("DROP TABLE IF EXISTS %s", fqTableName)
+		slog.Debug("Dropping temporary table", slog.String("sql", sqlCommand))
+		_, err := dwh.Exec(sqlCommand)
```
Comment (nathan-artie) [inline lib/destination/ddl/ddl.go:28]: Could combine with the following line.

**Review by nathan-artie — APPROVED** [review]

File: lib/destination/ddl/ddl.go (hunk 2 above)
Comment (Tang8330) [reply lib/destination/ddl/ddl.go]: Nice

**Pull request merged**